src/js/controller/controller.js

```diff
@@ -918,6 +918,21 @@ Object.assign(Controller.prototype, {
             return _setItem(_model.get('item'));
         };
 
+        this.setPlaylistem = function (index, item) {
+            const state = _model.get('state');
+            const selectedItemPlaying = index === _model.get('item') && state === 'playing';
```

Comment:
Just checking 'playing' isn't enough. The item is active when the player state is paused and buffering as well.

But this doesn't mean we shouldn't replace the item. It just means we can't reload it;

- Always replace the item
- When idle, if the item being replaced is the current item, call `setItem` to update the player state so that when playback starts the new item is played. This should also kick off preloading at the correct `item.startTime`.